src/processTargets/marks/DecoratedSymbolStage.ts

```diff
@@ -0,0 +1,32 @@
+import { DecoratedSymbolMark, Target } from "../../typings/target.types";
+import { ProcessedTargetsContext } from "../../typings/Types";
+import { getTokenContext } from "../modifiers/scopeTypeStages/TokenStage";
+import { MarkStage } from "../PipelineStages.types";
+
+export default class implements MarkStage {
+  constructor(private modifier: DecoratedSymbolMark) {}
+
+  run(context: ProcessedTargetsContext): Target[] {
+    const token = context.hatTokenMap.getToken(
+      this.modifier.symbolColor,
+      this.modifier.character
+    );
+    if (token == null) {
+      throw new Error(
+        `Couldn't find mark ${this.modifier.symbolColor} '${this.modifier.character}'`
+      );
+    }
+    const target = {
+      editor: token.editor,
+      contentRange: token.range,
+      isReversed: false,
+    };
+    return [
+      {
+        ...target,
+        ...getTokenContext(target),
+        scopeType: "token",
```

Comment:
how come this one isn't included in the output of `getTokenContext`?

I'm really tempted to switch to the `new Token(...)` setup.  All the object spreading / pojos are feeling like they'll be pretty tough to maintain

Reply:
Because the token context is used for other types as well. Decorated mark for example

Reply:
I'm going to rename it to something more clear